gnd/src/main/java/com/google/android/gnd/ui/map/MapAdapter.java

```diff
@@ -70,6 +70,12 @@
   /** Update map pins shown on map. */
   void setMapPins(ImmutableSet<MapPin> pins);
 
+  /** Get current map type. */
+  int getMapType();
+
+  /** Update map type. */
+  void setMapType(int mapType);
```

Comment:
Instead of using `int` here directly should we abstract away the particular map types of a given Map impl and introduce an enum and some conversion functions to allow us to set the types regardless of the underlying adapter?

@gino-m wdyt?

e.g. on the adapter we'd have some FN setMapType(TypeENUM) instead of int.

No need to do this now.

Reply:
It may make sense to define an enum at the interface level with a superset of basemap types, yes. Also +1 that it can be done in a follow up (please add a TODO)?

Also, somewhere we should clarify that Map Type refers to the basemap shown below map features and offline satellite imagery - they're called "map styles" in Mapbox and "basemaps" in Leaflet.

Reply:
Thanks @scolsen and @gino-m! Done